olp-cpp-sdk-core/include/olp/core/thread/ExecutionContext.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2022 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <utility>
+
+#include <olp/core/client/ApiError.h>
+#include <olp/core/client/CancellationContext.h>
+
+namespace olp {
+namespace thread {
+
+using FailedCallback = std::function<void(client::ApiError)>;
+using CancelFuncType = std::function<void()>;
+
+/// Handles the cancellation, final, and failed mechanisms.
+class CORE_API ExecutionContext {
+  using ExecuteFuncType = std::function<client::CancellationToken()>;
+
+ public:
+  /// A default contructor, initializes the `ExecutionContextImpl` instance.
+  ExecutionContext();
+
+  /**
+   * @brief Gets the error information.
+   *
+   * @return The `ApiError` instance containing the error information.
+   */
+  const client::ApiError GetError() const;
+
+  /**
+   * @brief Sets the error that is returned in the `Finally` method of
+   *  `Continuation`.
+   *
+   * @param error The `ApiError` instance containing the error information.
+   */
+  void SetError(client::ApiError error);
+
+  /**
+   * @brief Checks whether the error is set.
```

Comment:
Please explain what makes a failed `ExecutionContext`. Is it a call to `SetError()`?

Reply:
Removed it, found another way to provide an error for Continuation